Update sp-serveroption-transact-sql.md

VladDBA opened the pull request:
Hi SQL Server Docs Team,

I've done the following changes:
- Added missing option `name`, and its description, for `@optname`. 
- Added code example of using `@optname='name'` to rename a linked server connection.
- Corrected the data type and size of the `@optvalue` parameter.

Best regards,

Vlad

VladDBA committed:
Update sp-serveroption-transact-sql.md

Added missing option `name` for `@optname` and its description.
Added code example of using `@optname='name'` to rename a linked server connection
Corrected the data type and size of the `@optvalue` parameter.

VladDBA committed:
Update sp-serveroption-transact-sql.md

Court72 commented:
@markingmyname

Can you review the proposed changes? 

When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

rwestMSFT commented:
@VladDBA, I'm in the process of refreshing this entire folder, so this is helpful, thank you. We'll incorporate your changes into a future PR.

VladDBA committed:
Update sp-serveroption-transact-sql.md

Added missing option `provider string`, and its description, for parameter `@optname`.
Also fixed a small typo that I've introduced previously :)

VladDBA commented:
Hi @rwestMSFT, 

I've realized that there was another missing option called `provider string` for the `@optname` parameter, so I added it and its description.
Also fixed a small typo that I've introduced with this PR. Sorry about that :)

rwestMSFT commented:
@VladDBA Our internal PR is still in progress, so it's no big deal to adopt the additional changes. Thanks!
